rust/datafusion/src/sql/planner.rs

```diff
@@ -894,6 +908,131 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             ))),
         }
     }
+
+    fn sql_interval_to_literal(
+        &self,
+        value: &String,
+        leading_field: &Option<DateTimeField>,
+        leading_precision: &Option<u64>,
+        last_field: &Option<DateTimeField>,
+        fractional_seconds_precision: &Option<u64>,
+    ) -> Result<Expr> {
+        if leading_field.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with leading_field {:?}",
+                leading_field
+            )));
+        }
+
+        if leading_precision.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with leading_precision {:?}",
+                leading_precision
+            )));
+        }
+
+        if last_field.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with last_field {:?}",
+                last_field
+            )));
+        }
+
+        if fractional_seconds_precision.is_some() {
+            return Err(DataFusionError::NotImplemented(format!(
+                "Unsupported Interval Expression with fractional_seconds_precision {:?}",
+                fractional_seconds_precision
+            )));
+        }
+
+        // Should we resort parts on overflow?
+        let resort_parts = |days: i32, seconds: f32| -> (i32, f32) { (days, seconds) };
+
+        let calculate_from_part =
+            |interval_period_str: &str, interval_type: &str| -> Result<(i32, f32)> {
+                let interval_period = match interval_period_str.parse::<f32>() {
```

Comment:
I think this should be `i32` as well as the other types `f32`?

Reply:
I am using integer to support:

```
SELECT INTERVAL '0.5 day';
```

```
0 years 0 mons 0 days 12 hours 0 mins 0.00 secs
```

Reply:
Didn't realize that was possible. What about rounding errors (and rounding in general)?

Reply:
I dont have any ideas, how it should be correct, but anyway using `0.5 DAY` or `0.5 HOUR` will be rly usefull and works as expected.

Reply:
I am not sure whether rounding errors are going to be a problem, I could imagine some calculation might cause for example the number of milliseconds to be off for non-trivial fractions?
In that case we should use some decimal type instead of `f32` in order to avoid those rounding errors.

Reply:
@Dandandan 👍  I can place a comment in this place with // @todo, when I will finish with https://github.com/apache/arrow/pull/9232 I can reuse Decimal128 in this place. Is it ok? Thanks